📊 Adding UN IGME dataset and long-run child mortality data

spoonerf opened the pull request:
A combination of two child mortality sources to create a long-run indicator. I'm not sure I am handling the metadata correctly, the resulting variable at the end has two producers, but I can't see this reflected in the [datapage preview](http://staging-site-fiona/admin/datapage-preview/810791). 

Would it make more sense to manually curate the metadata for this indicator and condense the information into one producer?

pabloarosado commented on etl/steps/data/garden/gapminder/2023-09-18/under_five_mortality.excluded_countries.json:
```diff
@@ -0,0 +1,2 @@
+[
```
It'd be better to remove the file (and its mention in the code) instead of leaving it empty.

pabloarosado commented on snapshots/gapminder/2023-09-18/under_five_mortality.xlsx.dvc:
```diff
@@ -0,0 +1,30 @@
+meta:
+  origin:
+    # Data product / Snapshot
+    title: Child Mortality Rate (Gapminder)
```
The title should not mention the producer. Shouldn't the data product be `Child mortality rate under age five`?

pabloarosado commented on snapshots/gapminder/2023-09-18/under_five_mortality.xlsx.dvc:
```diff
@@ -0,0 +1,30 @@
+    # Data product / Snapshot
+    title: Child Mortality Rate (Gapminder)
+    description: |-
+      Gapminder’s historic estimates compiled by Klara Johansson and Mattias Lindgren.
```
This description doesn't say what the dataset contains. I'd instead move this to citation_full, and write here a brief line about the content of the dataset.

pabloarosado commented on snapshots/gapminder/2023-09-18/under_five_mortality.xlsx.dvc:
```diff
@@ -0,0 +1,30 @@
+      Gapminder’s historic estimates compiled by Klara Johansson and Mattias Lindgren.
+    date_published: 2015-01-01
+    version_producer: v7
+    title_snapshot: Child Mortality Rate (Gapminder)
```
If you want to specify that you are extracting an old version of the data product, I suppose you could say `Child mortality rate under age five v7`.

pabloarosado commented on snapshots/gapminder/2023-09-18/under_five_mortality.xlsx.dvc:
```diff
@@ -0,0 +1,30 @@
+
+    # Citation
+    producer: Gapminder
+    citation_full: Child Mortality Rate, v7, Gapminder (2015)
```
You should include here more details about the authors, something like:
"Compiled and documented by Klara Johansson and Mattias Lindgren from many sources but mainly based on..."

pabloarosado commented on snapshots/gapminder/2023-09-18/under_five_mortality.xlsx.dvc:
```diff
@@ -0,0 +1,30 @@
+    # Citation
+    producer: Gapminder
+    citation_full: Child Mortality Rate, v7, Gapminder (2015)
+    attribution_short: Gapminder (2015)
```
Attribution short should simply be `Gapminder`, without the year.

pabloarosado commented on snapshots/gapminder/2023-09-18/under_five_mortality.xlsx.dvc:
```diff
@@ -0,0 +1,30 @@
+
+    # License
+    license:
+      name: CC BY 4.0# License (same as origin.license, for backwards compatibility)
```
License name should be "CC BY 4.0". I think there may be a bug in wizard, that adds the comment at the end.

lucasrodes replied:
Yes, there was an issue with `wizard` templates. This was fixed and should work again. 

For this file I recommend having

```yaml
...
    # License
    license:
      name: CC BY 4.0
  # License (same as origin.license, for backwards compatibility)
  license:
    name: CC BY 4.0
...
```

(basically, (i) remove the comment from line 23 and copy-paste the `license` block in line 24 with one indent less)

Marigold replied:
Why do we need `meta.license` for backwards compatibility? If there's some technical reason, then we should fix it.

pabloarosado replied:
Last time I tried to run a snapshot step without the additional meta.license, it failed. Yes, if that's still the case, we should definitely fix that.

Marigold replied:
@pabloarosado I removed the license, and it didn't fail, though it raised a warning about the missing license (that I'm removing in #1638 ). If run into it again, could you tell me how to replicate it?

pabloarosado commented on etl/steps/data/garden/gapminder/2023-09-18/under_five_mortality.meta.yml:
```diff
@@ -0,0 +1,114 @@
+      under_five_mortality:
+        title: Under five mortality
+        origins:
+          - producer: Gapminder
```
You don't need to define origins here. They should be propagated automatically. If you did it because you couldn't pinpoint where the propagation failed, I can help you out.

pabloarosado commented on etl/steps/data/garden/gapminder/2023-09-18/under_five_mortality.meta.yml:
```diff
@@ -0,0 +1,114 @@
+  under_five_mortality:
+    title: Child Mortality Rate (Gapminder)
+    description: |
+      "This data has been compiled by Klara Johansson and Mattias Lindgren (Gapminder) from a selection of sources:
```
This description should rather be citation_full. If it refers only to one of the origins, then it should be defined at the snapshot.

pabloarosado commented on etl/steps/data/garden/gapminder/2023-09-18/under_five_mortality.meta.yml:
```diff
@@ -0,0 +1,114 @@
+          - name: CC BY 4.0
+            url: null
+        unit: deaths per 1,000 live births
+      source:
```
You don't need to define sources anymore.

pabloarosado commented on etl/steps/data/garden/un/2023-08-16/igme.py:
```diff
@@ -36,14 +36,13 @@ def run(dest_dir: str) -> None:
     for col in tb.columns[2:]:
         unit = col.split("-")[1]
-        tb[col].metadata = VariableMeta(unit=unit)
+        tb[col].metadata = VariableMeta(unit=unit.lower().strip())
```
Hmm I suppose that when you do this, you are rewriting the metadata of the variable, without adding all other fields (e.g. origins).
Why not doing
```
tb[col].metadata.unit = unit.lower().strip()
```
?

pabloarosado commented on etl/steps/data/garden/un/2023-08-29/long_run_child_mortality.meta.yml:
```diff
@@ -0,0 +1,96 @@
+dataset:
+  title: Long-run child mortality (Gapminder; UN IGME)
```
We are moving away from adding producer names in dataset titles, but for now you can keep it if you think it's convenient.

pabloarosado commented on etl/steps/data/garden/un/2023-08-29/long_run_child_mortality.meta.yml:
```diff
@@ -0,0 +1,96 @@
+dataset:
+  title: Long-run child mortality (Gapminder; UN IGME)
+  origins:
```
Again, you shouldn't need to define origins here, they should already be present from the snapshot.

pabloarosado commented on etl/steps/data/garden/un/2023-08-29/long_run_child_mortality.py:
```diff
@@ -0,0 +1,88 @@
+    tb_igme = ds_igme["igme"].reset_index()
+
+    # Select out columns of interest.
+    cols = [
```
An easier implementation would be to define cols as a dictionary:
```
columns = {
    "country": "country",
    "year": "year",
    "observation_value_deaths_per_1_000_live_births_under_five_mortality_rate_both_sexes_all_wealth_quintiles": "under_five_mortality",
} 
tb_igme = tb_igme[list(columns)].rename(columns=columns, errors="raise")
```

pabloarosado commented on etl/steps/data/garden/un/2023-08-29/long_run_child_mortality.py:
```diff
@@ -0,0 +1,88 @@
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(dest_dir, tables=[tb_combined, tb_combined_sel], check_variables_metadata=True)
+    ds_garden.update_metadata(metadata_path=paths.metadata_path)
```
I don't think you need to use update_metadata. This should happen automatically inside `create_dataset`.

pabloarosado reviewed (changes requested):
There are several things that would be good to change with respect to metadata. If you need help, please let me know.

spoonerf commented:
> There are several things that would be good to change with respect to metadata. If you need help, please let me know.

Thanks @pabloarosado! I think I have addressed your points now; thanks for taking the time to review. 

I am still not certain it is working quite as expected e.g. [the data page preview](http://staging-site-fiona/admin/datapage-preview/810791) features mostly Gapminder, but I think that might be because the UN IGME data was imported with the old style metadata?
